DAL/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,65 @@
+﻿using DAL.Models;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DAL.Repositories
+{
+    internal abstract class BaseRepository<T> where T : BaseModel
+    {
+        private InternshipDbContext _internshipDbContext;
+        private DbSet<T> _dbSet;
+
+        public BaseRepository(InternshipDbContext internshipDbContext)
+        {
+            _internshipDbContext = internshipDbContext;
+            _dbSet = _internshipDbContext.Set<T>();
+        }
+
+        public virtual List<T> GetAll()
+        {
+            return _dbSet.ToList();
+        }
+
+        public virtual T Get(long id)
+        {
+            return _dbSet.SingleOrDefault(x => x.Id == id);
+        }
+
+        public virtual void Save(T model)
+        {
+            if (model.Id > 0)
+            {
+                _dbSet.Update(model);
+            }
+            else
+            {
+                _dbSet.Add(model);
+            }
+            _internshipDbContext.SaveChanges();
```

Comment:
I suppose in this case we cannot do atomic transaction: for example in our business logic we need one time to do few operation: delete from candidate table and create new row in the topic. If after deleting will throw some exception then in result will be deleting row from candidate table and without new row in the topic table. Without transactions we shouldn't do it. In this case it needs separate method for saving data.